src/libstd/process.rs

```diff
@@ -0,0 +1,835 @@
+    /// Format the program and arguments of a Command for display. Any
+    /// non-utf8 data is lossily converted using the utf8 replacement
+    /// character.
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        try!(write!(f, "{:?}", self.inner.program));
+        for arg in &self.inner.args {
+            try!(write!(f, " '{:?}'", arg));
+        }
+        Ok(())
+    }
+}
+
+impl AsInner<CommandImp> for Command {
+    fn as_inner(&self) -> &CommandImp { &self.inner }
+}
+
+impl AsInnerMut<CommandImp> for Command {
+    fn as_inner_mut(&mut self) -> &mut CommandImp { &mut self.inner }
+}
+
+fn setup_io(io: &Stdio, fd: libc::c_int, readable: bool)
+            -> IoResult<(Option<FileDesc>, Option<FileDesc>)>
+{
+    use self::Stdio::*;
+    Ok(match *io {
+        Null => {
+            (None, None)
+        }
+        Inherit => {
+            (Some(FileDesc::new(fd, false)), None)
+        }
+        Capture => {
+            let (reader, writer) = try!(unsafe { sys::os::pipe_shim() });
+            if readable {
+                (Some(reader), Some(writer))
+            } else {
+                (Some(writer), Some(reader))
+            }
+        }
+    })
+}
+
+/// The output of a finished process.
+#[derive(PartialEq, Eq, Clone)]
+pub struct Output {
+    /// The status (exit code) of the process.
+    pub status: ExitStatus,
+    /// The data that the process wrote to stdout.
+    pub output: Vec<u8>,
+    /// The data that the process wrote to stderr.
+    pub error: Vec<u8>,
+}
+
+/// Describes what to do with a standard io stream for a child process.
+// FIXME: should be an extensible enum
+#[derive(Clone)]
+pub enum Stdio {
+    /// A new pipe should be arranged to connect the parent and child processes.
+    Capture,
+
+    /// The child inherits from the corresponding parent descriptor.
+    Inherit,
+
+    /// This stream will be ignored. This is the equivalent of attaching the
+    /// stream to `/dev/null`
+    Null,
+}
+
+/// Describes the result of a process after it has terminated.
+#[derive(PartialEq, Eq, Clone, Copy, Debug)]
+pub enum ExitStatus {
+    /// Normal termination with an exit code.
+    Code(i32),
+
+    /// Termination by signal, with the signal number.
+    ///
+    /// Never generated on Windows.
+    Signal(i32),
+}
+
+impl ExitStatus {
+    /// Was termination successful? Signal termination not considered a success,
+    /// and success is defined as a zero exit status.
+    pub fn success(&self) -> bool {
+        return *self == ExitStatus::Code(0);
+    }
+}
+
+#[stable(feature = "rust1", since = "1.0.0")]
+impl fmt::Display for ExitStatus {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            ExitStatus::Code(code) =>  write!(f, "exit code: {}", code),
+            ExitStatus::Signal(code) =>  write!(f, "signal: {}", code),
+        }
+    }
+}
+
+impl Child {
+    /// Sends a signal to this child forcing it to exit. This is equivalent to
```

Comment:
“Sends a signal” exposes a certain implementation detail which should not be exposed. Especially given there’s [TerminateProcess](https://msdn.microsoft.com/en-us/library/windows/desktop/ms686714%28v=vs.85%29.aspx).